src/ImageSharp/Memory/DiscontiguousBuffers/MemoryGroup{T}.cs

```diff
@@ -241,38 +245,77 @@ public static MemoryGroup<T> Wrap(params IMemoryOwner<T>[] source)
-        /// Swaps the contents of 'target' with 'source' if the buffers are allocated (1),
-        /// copies the contents of 'source' to 'target' otherwise (2).
-        /// Groups should be of same TotalLength in case 2.
-        /// </summary>
-        public static bool SwapOrCopyContent(MemoryGroup<T> target, MemoryGroup<T> source)
+        [MethodImpl(InliningOptions.ShortMethod)]
+        public unsafe Span<T> GetRowSpanCoreUnsafe(int y, int width)
         {
-            if (source is Owned ownedSrc && ownedSrc.Swappable &&
-                target is Owned ownedTarget && ownedTarget.Swappable)
+            switch (this.memoryGroupSpanCache.Mode)
             {
-                Owned.SwapContents(ownedTarget, ownedSrc);
-                return true;
-            }
-            else
-            {
-                if (target.TotalLength != source.TotalLength)
+                case SpanCacheMode.SingleArray:
+                {
+#if SUPPORTS_CREATESPAN
+                    ref byte b0 = ref MemoryMarshal.GetReference<byte>(this.memoryGroupSpanCache.SingleArray);
+                    ref T e0 = ref Unsafe.As<byte, T>(ref b0);
+                    e0 = ref Unsafe.Add(ref e0, y * width);
+                    return MemoryMarshal.CreateSpan(ref e0, width);
+#else
+                    return MemoryMarshal.Cast<byte, T>(this.memoryGroupSpanCache.SingleArray).Slice(y * width, width);
+#endif
+
+                }
+
+                case SpanCacheMode.SinglePointer:
                 {
-                    throw new InvalidMemoryOperationException(
-                        "Trying to copy/swap incompatible buffers. This is most likely caused by applying an unsupported processor to wrapped-memory images.");
+                    void* start = Unsafe.Add<T>(this.memoryGroupSpanCache.SinglePointer, y * width);
+                    return new Span<T>(start, width);
                 }
 
-                source.CopyTo(target);
-                return false;
+                case SpanCacheMode.MultiPointer:
+                {
+                    this.GetMultiBufferPosition(y, width, out int bufferIdx, out int bufferStart);
+                    void* start = Unsafe.Add<T>(this.memoryGroupSpanCache.MultiPointer[bufferIdx], bufferStart);
+                    return new Span<T>(start, width);
+                }
+
+                default:
+                {
+                    this.GetMultiBufferPosition(y, width, out int bufferIdx, out int bufferStart);
+                    return this[bufferIdx].Span.Slice(bufferStart, width);
+                }
             }
         }
 
+        /// <summary>
+        /// Returns the slice of the buffer starting at global index <paramref name="start"/> that goes until the end of the buffer.
+        /// </summary>
+        public Span<T> GetRemainingSliceOfBuffer(long start)
+        {
+            int bufferIdx = (int)(start / this.BufferLength);
+            int bufferStart = (int)(start % this.BufferLength);
```

Comment:
Could use `Math.DivRem` to do this in one operation?

Reply:
When you look at the x86 disassembly, then it's expected -- sharplab sometime bites me too.
[Looking at x64](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYQYG8aHunHjykDAJYA7DAwCCACgA2EEQHMGuDNigY0DOYq0xFGABYBKDlx7nR47AwC8DKZaNSVa8QHpd+4wG4z57pYMwLb2js6q6gwApJ4Khka+1P48xADsDNgA1MCJ5gC+fgyF9EwCwmIMAEKy8kou6praSjJ6ccamSclatRkhALLYhgB0ACJCAG4ASjD44a6NrYaaEACu4k1BCYXmaaFiRlkO+zmFBdR5QA=) it's better.

Reply:
Indeed, thanks!

Reply:
Multi-buffer case went from 45ns to 38ns! (0d3bd57)